test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
@alshakero Is it requirement per-se, or just an implementation limitation?

@warpech Don't you think it could be confusing for our developers that:
```js
ProxyB.proxifiedObject = {foo: 1};
ProxyA.newValue = ProxyB.proxifiedObject;
ProxyA.newValue === ProxyB.proxifiedObject // false

ProxyA.newValue.add = 1;
ProxyB.proxifiedObject.add // undefined
```
?